scootapi/client/smoke_test_cmd.go

```diff
@@ -45,121 +45,94 @@ func (c *Client) runSmokeTest(cmd *cobra.Command, args []string) error {
 			return err
 		}
 	}
-	// run a bunch of concurrent jobs and track their status
-	ch := make(chan jobAndStatus)
-	timedOutJobs := make(chan error)
-	var wg sync.WaitGroup
-	errCh := make(chan error, numTasks)
+	jobs := make(map[string]*scoot.JobStatus)
 	for i := 0; i < numTasks; i++ {
-		wg.Add(1)
-		go func() {
-			err := c.generateAndRunJob(timeout, ch)
-			if err != nil {
-				timedOutJobs <- err
-				errCh <- err
-				fmt.Println(err)
-			}
-			wg.Done()
-		}()
-	}
-
-	jobStatusMap := make(map[string]scoot.Status)
-	ticker := time.NewTicker(time.Millisecond)
-	timeouts := 0
-Loop:
-	for {
-		select {
-		case <-timedOutJobs:
-			timeouts++
-		case <-ticker.C:
-			// job id's grouped by status
-			statusJobMap := make(map[scoot.Status][]string)
-			jobAndStatus := <-ch
-			jobStatusMap[jobAndStatus.job] = jobAndStatus.status
-			for job, status := range jobStatusMap {
-				// populate statusJobMap
-				statusJobMap[status] = append(statusJobMap[status], job)
-			}
-			fmt.Println(timeouts, "jobs have timed out")
-			for status, jobs := range statusJobMap {
-				sort.Sort(sort.StringSlice(jobs))
-				fmt.Println(status, ":", jobs)
-				// if all jobs are completed, break loop
-				if len(statusJobMap[scoot.Status_COMPLETED])+len(statusJobMap[scoot.Status_ROLLED_BACK])+timeouts == numTasks {
-					ticker.Stop()
-					break Loop
-				}
-			}
+		id, err := c.generateAndStartJob()
+		if err != nil {
+			return err
 		}
+		jobs[id] = nil
 	}
 
-	wg.Wait()
-
-	// if any errors were logged return an error
-	select {
-	case err := <-errCh:
-		return err
-	default:
-		return nil
-	}
+	return c.waitForJobs(jobs, timeout)
 }
 
-func (c *Client) generateAndRunJob(timeout time.Duration, ch chan jobAndStatus) error {
+func (c *Client) generateAndStartJob() (string, error) {
 	client, err := c.Dial()
-
 	if err != nil {
-		return err
+		return "", err
 	}
-
 	rng := rand.New(rand.NewSource(time.Now().UnixNano()))
 
 	// We just want the JobDefinition here Id doesn't matter
 	job := testhelpers.GenJobDefinition(rng)
 	jobId, err := client.RunJob(job)
+	return jobId.ID, err
+}
 
-	// Error Enqueuing Job
-	if err != nil {
-		switch err := err.(type) {
-		case *scoot.InvalidRequest:
-			return fmt.Errorf("Invalid Request: %v", err.GetMessage())
-		default:
-			return fmt.Errorf("Error running job: %v %T", err, err)
+func (c *Client) waitForJobs(jobs map[string]*scoot.JobStatus, timeout time.Duration) error {
+	end := time.Now().Add(timeout)
```

Comment:
yes the original idea behind the timeout is the smoketest as a whole should run for "X seconds" before timing out.  In-case there is a bug and it doesn't just wait forever